### ISSUE TITLE
"Wide" block alignment overflows on thinner desktop widths (922px to about 1200px)

### ISSUE THREAD

**Issue by ZebulanStanphill** [brainstormforce/astra#3405]
### Describe the bug:
I should be able to view a page with "Wide"-aligned Gutenberg blocks and not notice any horizontal overflow when using viewports from 922px to about 1200px wide. However, this is not the case. Because the styles for the "Wide" block alignment use hardcoded `-100px` horizontal margins, they cause the edges of the block to go off-screen at certain thin desktop viewport widths that are too wide for the mobile styles (which disable the "Wide" alignment styling entirely) to kick in.

### To reproduce:
1. Create and view a page with the following content:
```html
<!-- wp:columns {"align":"wide","backgroundColor":"ast-global-color-0"} -->
<div class="wp-block-columns alignwide has-ast-global-color-0-background-color has-background"><!-- wp:column {"width":"50%"} -->
<div class="wp-block-column" style="flex-basis:50%"><!-- wp:paragraph {"textColor":"ast-global-color-5"} -->
<p class="has-ast-global-color-5-color has-text-color">this is a Paragraph inside of a Wide-aligned Group.</p>
<!-- /wp:paragraph --></div>
<!-- /wp:column -->

<!-- wp:column {"width":"50%"} -->
<div class="wp-block-column" style="flex-basis:50%"><!-- wp:paragraph -->
<p>another paragraph</p>
<!-- /wp:paragraph --></div>
<!-- /wp:column --></div>
<!-- /wp:columns -->
```
2. Shrink your viewport width to something thin, but not so thin that the mobile styles kick in. About `930px` should make the issue clearly visible.

### Suggested solution:
Change the horizontal margins for "Wide"-aligned blocks to something like `max(calc(50% - 50vw + 20px), -100px)`; this clamps the negative margins to prevent horizontal overflow. In case you're wondering, the `20px` in the `calc` is the standard horizontal padding used by `.ast-container`. This makes the wide blocks have (almost, see next paragraph) the same spacing on the edges as normal-aligned blocks at the mobile viewports.

I've tested this solution locally and it works great, with the one minor annoyance that the horizontal spacing around the "Wide" blocks appears ever-so-slightly less than what it should be, since `vw` units don't take into account the width of the browser's scrollbar. (Why, CSS, why?) But this is barely noticeable, and if it really bothers you, it is possible to fix by [using JS to calculate the scroll bar width and save it to a CSS custom property](https://stackoverflow.com/a/57748743).

**TL;DR:** Replace
```css
.ast-no-sidebar .entry-content .alignwide {
  margin-left: -100px;
  margin-right: -100px;
  max-width: unset;
  width: unset;
}
```
with
```css
:root {
  /* Using a CSS custom property just makes it easier for people to customize. */
  --alignwide-negative-margin: max(calc(50% - 50vw + 20px), -100px);
}
.ast-no-sidebar .entry-content .alignwide {
  margin-left: var(--alignwide-negative-margin);
  margin-right: var(--alignwide-negative-margin);
  max-width: unset;
  width: unset;
}
```

### Screenshots:
![image](https://user-images.githubusercontent.com/19592990/136128917-31f19f8a-19ee-4d3a-ba21-9d0fbdce363e.png)
![image](https://user-images.githubusercontent.com/19592990/136128949-5db8565b-ebac-4bbb-87ae-76712d7f2f16.png)

### Isolating the problem:
- [x] This bug happens with no other plugins activated
- [x] I can reproduce this bug consistently using the steps above

### WordPress version:
5.8.1

### Astra Theme version:
3.7.3

**Comment by imnavanath** [comment]
Hi @ZebulanStanphill,

Thanks for reporting it.

As you know and we already discussed in another issue, we have already started improving editor & frontend UI for Gutenberg blocks based on their layouts and with few new improvements.

I verified current issue in `next-release` it seems to be fix now onwards, as we already removed this CSS for improving layout design by handling backward for existing sites.

See details here - 
Replicated issue in current version - https://share.bsf.io/12uPppQO
next-release fix after adding filter - https://share.bsf.io/yAu0GGng 

Some finishing changes are still in WIP. But I would request you to please verify the fix in the branch `next-release`. 
While reviewing it please make sure that you will use following filter in your child theme - 
`add_filter( 'astra_get_option_improve-gb-editor-ui', '__return_true' );`

Thanks.

**Comment by ZebulanStanphill** [comment]
@imnavanath 
I just realized that my example HTML had incorrect text. It says "inside a Wide-aligned Group" when it should say "inside a Wide-aligned Columns". Whoops. I've updated the example HTML to correct this mistake.

Anyway, I built the theme from `next-release` and added the filter as you suggested. But it looks like the issue is only "fixed" because wide alignment is now just not working at all, regardless of viewport width. I assume that's not intentional?
![image](https://user-images.githubusercontent.com/19592990/136277770-ece62217-cbe1-4043-84b3-1c95d75fb4fa.png)
![image](https://user-images.githubusercontent.com/19592990/136278243-2b6afa3e-297f-4df5-adcf-e894a550ca11.png)
